AutomatedTesting/Gem/PythonTests/Atom/atom_utils/material_canvas_utils.py

```diff
@@ -10,101 +10,101 @@
+
+
+class Node(object):
+    """
+    Represents a node inside of a Graph (required).
+    """
+
+    def __init__(self, node_object: object, node_id: int, node_name: str, graph_id: math.Uuid, position: math.Vector2):
+        self.node_object = node_object
+        self.node_id = node_id
+        self.node_name = node_name
+        self.graph_id = graph_id
+        self.position = position
+        self.outbound_slot = graph.GraphModelSlotId('outPosition')
+        self.inbound_slot = graph.GraphModelSlotId('inPositionOffset')
+
+    def connect_to_node_outbound_slot(self, target_node: Node) -> object:
+        """
+        Adds a new connection between this node's inbound slot to target_node's outbound slot.
+        :param target_node: A Node() object for the target node to connect to.
+        :return: azlmbr.object.PythonProxyObject representing a C++ AZStd::shared_ptr<Connection> object.
+        """
+        return graph.GraphControllerRequestBus(
+            bus.Event, GraphControllerRequestBusEvents.ADD_CONNECTION_BY_SLOT_ID, self.graph_id,
+            self.node_object, self.inbound_slot, target_node, target_node.outbound_slot)
+
+    def connect_to_node_inbound_slot(self, target_node: Node) -> object:
+        """
+        Adds a new connection between this node's outbound slot to target_node's inbound slot.
+        :param target_node: A Node() object for the target node to connect to.
+        :return: azlmbr.object.PythonProxyObject representing a C++ AZStd::shared_ptr<Connection> object.
+        """
+        return graph.GraphControllerRequestBus(
+            bus.Event, GraphControllerRequestBusEvents.ADD_CONNECTION_BY_SLOT_ID, self.graph_id,
+            self.node_object, self.outbound_slot, target_node, target_node.inbound_slot)
+
+    def is_connected_to_node_outbound_slot(self, target_node: Node) -> bool:
+        """
+        Determines if self.node_object's inbound slot is connected to the target_node's outbound slot.
+        :param target_node: A Node() object for the target node to connect to.
+        :return: bool representing success (True) or failure (False)
+        """
+        return graph.GraphControllerRequestBus(
+            bus.Event, GraphControllerRequestBusEvents.ARE_SLOTS_CONNECTED, self.graph_id,
+            self.node_object, self.inbound_slot, target_node, target_node.outbound_slot)
+
+    def is_connected_to_node_inbound_slot(self, target_node: Node) -> bool:
+        """
+        Determines if self.node_object's outbound slot is connected to the target_node's inbound slot.
+        :param target_node: A Node() object for the target node to connect to.
+        :return: bool representing success (True) or failure (False)
+        """
+        return graph.GraphControllerRequestBus(
+            bus.Event, GraphControllerRequestBusEvents.ARE_SLOTS_CONNECTED, self.graph_id,
+            self.node_object, self.outbound_slot, target_node, target_node.inbound_slot)
```

Comment:
I would also move this to the graph class and use the signature that's already there on the bus. No reason to differentiate between inbound and outbound when they're doing the same thing.